lib/pagination/adapter.js

```diff
@@ -0,0 +1,35 @@
+const { invalidPageNumber, invalidLimitNumber } = require('./errors');
+
+exports.content = undefined;
+exports.page = undefined;
+exports.limit = undefined;
+exports.request = undefined;
+
+exports.createAdapter = (content, page, limit, req) => {
+  if (typeof page !== 'number' || page <= 0) throw invalidPageNumber('Invalid Page Number');
+  if (typeof limit !== 'number' || limit <= 0) throw invalidLimitNumber('Invalid Limit Number');
+
+  exports.content = content;
+  exports.page = page;
+  exports.limit = limit;
+  exports.request = req;
```

Comment:
Don't use `exports` for keeping state of adapter it's a little bit tricky, you could use an object directly